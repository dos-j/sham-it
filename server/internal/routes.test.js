const registerRoutes = require("./routes");

jest.mock("./expectations/routes", () => {
  const register = jest.fn(() => register.__routes);
  register.__routes = ["a", "b", "c"];

  return register;
});
const registerExpectations = require("./expectations/routes");

jest.mock("./healthcheck", () => () => "healthcheck");
jest.mock("../shamCore", () => {
  const shamCore = {
    getAllRequestsHandler: jest.fn(() => "getAllRequestsHandler"),
    getAllMatchersHandler: jest.fn(() => "getAllMatchersHandler"),
    getSingleMatcherHandler: jest.fn(() => "getSingleMatcherHandler"),
    createMatcherHandler: jest.fn(() => "createMatcherHandler"),
    deleteMatcherHandler: jest.fn(() => "deleteMatcherHandler"),
    resetHandler: jest.fn(() => "resetHandler"),
    shutdownHandler: jest.fn(() => "shutdownHandler")
  };
  return shamCore;
});
const shamCore = require("../shamCore");

describe("unit: internal/routes", () => {
  let routes;
  let utils;

  beforeEach(() => {
    utils = {
      internalRoute: jest.fn(
        (method, regex, handler) => `${method} - ${regex} - ${handler}`
      ),
      matcherStore: [],
      requestStore: [],
      server: { shutdown() {} }
    };

    Object.entries(shamCore).forEach(([, func]) => func.mockClear());

    routes = registerRoutes(utils);
  });

  test("it should register the expectation routes", () => {
    expect(registerExpectations).toHaveBeenCalledWith(utils);
  });

  test("it should return all of the internal routes", () => {
    expect(routes).toMatchSnapshot();
  });
});
