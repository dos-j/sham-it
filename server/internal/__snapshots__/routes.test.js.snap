// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit: internal/routes it should return all of the internal routes 1`] = `
Array [
  "a",
  "b",
  "c",
  "GET - /^\\\\/\\\\$health$/ - healthcheck",
  "GET - /^\\\\/\\\\$matchers$/ - getAllMatchersHandler",
  "GET - /^\\\\/\\\\$matchers\\\\/(.*)$/ - getSingleMatcherHandler",
  "POST - /^\\\\/\\\\$matchers$/ - createMatcherHandler",
  "DELETE - /^\\\\/\\\\$matchers\\\\/(.*)$/ - deleteMatcherHandler",
  "GET - /^\\\\/\\\\$requests$/ - getAllRequestsHandler",
  "POST - /^\\\\/\\\\$reset$/ - resetHandler",
  "POST - /^\\\\/\\\\$shutdown$/ - shutdownHandler",
]
`;
